server/configuration.go

```diff
@@ -1,47 +1,148 @@
+		logger.Error("Error:", err)
+	} else {
+		logger.Debug("Directory already exists: ", dirPath)
+	}
+}
+
+func createDefaultConfigFile(cfgFile string) {
+
+	var passwordNew string = utils.GenerateRandomString(12)
+
+	logger.Info("Creating default configuration file")
+
+	viper.SetDefault("server.host", "127.0.0.1")
+	viper.SetDefault("server.port", "2605")
+	viper.SetDefault("server.admin_user", "admin")
+	viper.SetDefault("server.admin_password", passwordNew)
+
+	viper.SetDefault("log.log_level", "INFO")
+	viper.SetDefault("log.log_file", "daredb.log")
+
+	viper.SetDefault("settings.data_dir", DATA_DIR)
+	viper.SetDefault("settings.settings_dir", SETTINGS_DIR)
+
+	viper.SetDefault("security.tls_enabled", false)
+	viper.SetDefault("security.cert_private", filepath.Join(SETTINGS_DIR, "cert_private"))
+	viper.SetDefault("security.cert_public", filepath.Join(SETTINGS_DIR, "cert_public"))
+
+	viper.WriteConfigAs(cfgFile)
+
+	fmt.Printf("\nIMPORTANT! Generate default password for admin on initial start. Store it securely. Password: %v\n\n", passwordNew)
+}
+
+func mappingEnvsToConfig() {
+
+	mapsEnvsToConfig["server.host"] = "DAREDB_HOST"
+	mapsEnvsToConfig["server.port"] = "DAREDB_PORT"
+	mapsEnvsToConfig["server.root_user_name"] = "DAREDB_USER"
+	mapsEnvsToConfig["server.root_user_password"] = "DAREDB_PASSWORD"
+
+	mapsEnvsToConfig["log.log_level"] = "DAREDB_LOG_LEVEL"
+	mapsEnvsToConfig["log.log_file"] = "DAREDB_LOG_FILE"
+
+	mapsEnvsToConfig["settings.data_dir"] = "DAREDB_DATA_DIR"
+	mapsEnvsToConfig["settings.base_dir"] = "DAREDB_BASE_DIR"
+	mapsEnvsToConfig["settings.settings_dir"] = "DAREDB_SETTINGS_DIR"
+
+	mapsEnvsToConfig["security.tls_enabled"] = "DARE_TLS_ENABLED"
+	mapsEnvsToConfig["security.cert_private"] = "DAREDB_CERT_PRIVATE"
+	mapsEnvsToConfig["security.cert_public"] = "DAREDB_CERT_PUBLIC"
+}
+
+func reReadConfigsFromEnvs() {
+	logger.Info("Re-reading configurations from environmental variables")
+	for key, value := range mapsEnvsToConfig {
+		valueFromEnv, ok := os.LookupEnv(value)
+		if ok {
+			fmt.Println(key, valueFromEnv)
+			viper.Set(key, valueFromEnv)
+		}
 	}
 }
 
-func getEnvOrDefault(key string, defaultValue interface{}) interface{} {
-	value := os.Getenv(key)
-	if value == "" {
-		return defaultValue
+func initDBDirectories() {
+
+	dbBaseDir, err := os.Getwd()
+	if err != nil {
+		logger.Error("Error in getting current working directory:", err)
+	}
+	os.Setenv("DAREDB_BASE_DIR", dbBaseDir)
+
+	createDirectory(filepath.Join(dbBaseDir, SETTINGS_DIR))
+	createDirectory(filepath.Join(dbBaseDir, viper.GetString("settings.data_dir")))
+}
+
+func PrintConfigsToConsole() {
+	fmt.Printf("Print all configs that were set\n")
+	for key, _ := range mapsEnvsToConfig {
+		fmt.Printf("Config value for for %v is: %v\n", key, viper.Get(key))
 	}
-	return value
 }
 
-func getEnvBooleanOrDefault(key string, defaultValue bool) bool {
-	value := os.Getenv(key)
-	if value == "" {
-		return defaultValue
+func Configure(cfgFile string) {
```

Comment:
As for the logger, I'd like to create a Configurator struct with its own methods. 
Do you think it's hard to handle all the conf params without viper but only with native features ?

Reply:
The problem is that someone will end writing a lot of boiler plate code to handle the config (e.g., handling the TOML format, saving configs to the file, reading from the file, assigning to the values from the config correct types, handling edge cases, etc.). That is the reason why a well-known and widespread package such as ```viper``` could be a timesaver.

Reply:
Ok. I will try to do my best in this sense

Reply:
I addressed this comment in the commit https://github.com/dmarro89/dare-db/pull/18/commits/ccec10b5c978a513b2594ea2817199657d9e1b1b

Does it fits?

Important: didn't touch logger for now